### ISSUE TITLE
pinned tab order missing after restart

### ISSUE THREAD

**Issue by darkdh** [brave/browser-laptop#7091]
STR:
1. Open site A, B, C 
2. Pinned site A, B, C
3. Move pinned tab A to the place between B & C
4. Relaunch Brave
5. **Pinned tab order is not preserved**

cc @bbondy, @bsclifton

**Comment by akaihola** [comment]
My pinned tabs don't preserve order on version 0.14.0 rev 61b6b6b. Was this fix still not included there, or has there been a regression after version 0.13.3?
